src/artm/core/processor.cc

```diff
@@ -467,13 +541,19 @@ InferThetaAndUpdateNwtSparse(const ModelConfig& model_config, const Batch& batch
     for (float& value : values) value *= batch_weight;
     nwt_writer->Store(w, token_id[w], values);
   }
+
+  CreateThetaCacheEntry(new_cache_entry_ptr, theta_matrix, batch, part->model_name(), model_increment, topics_count);
 }
 
 static void
 InferPtdwAndUpdateNwtSparse(const ModelConfig& model_config, const Batch& batch, float batch_weight, const Mask* mask,
                             const InstanceSchema& schema, const CsrMatrix<float>& sparse_ndw,
                             const ::artm::core::PhiMatrix& p_wt, DenseMatrix<float>* theta_matrix,
-                            NwtWriteAdapter* nwt_writer, util::Blas* blas) {
+                            NwtWriteAdapter* nwt_writer, util::Blas* blas,
+                            std::shared_ptr<DataLoaderCacheEntry> new_cache_entry_ptr = nullptr,
```

Comment:
Using shared pointer is a bad practice. Its better to replace it with native pointer. This is because caller can always get a native pointer from ``std::shared_ptr``, but not the other way around. There are cases when passing ``std::shared_ptr`` is appropriate, but they should be related to some kind of memory management (like passing an object to and from ProcessorQueue in BigARTM).